Add a merge strategy

mdevlamynck opened the pull request:
Fixes #72.

Adds a `--use-merge-strategy` option. It will cause marge to use `git merge` instead of `git rebase` when updating the merge request branch. This option conflicts with `--add-tested`.

I struggled a bit trying to find correct generic terms englobing both merge and rebase. Depending on the context I used update and fuse. There may be better ones.

jcpetruzza commented on marge/app.py:
```diff
@@ -150,6 +150,15 @@ def regexp(s):
+            '(enable this is you use git merge as\n'
+            'git tends to misbehave when both are used)\n'
+        ),
+    )
```
I see you later renamed this to `--use-merge-strategy`, which I like better.... The rest still applies.

jcpetruzza commented on marge/job.py:
```diff
@@ -253,6 +254,127 @@ def during_merge_embargo(self):
+        changes_pushed = True
+    except git.GitError:
+        if not branch_merged:
+            raise CannotMerge('got conflicts while rebasing, your problem now...')
```
I see this was later fixed!

mdevlamynck commented on marge/git.py:
```diff
@@ -59,6 +59,38 @@ def tag_with_trailer(self, trailer_name, trailer_values, branch, start_commit):
         return self.get_commit_hash()
 
+    def merge(self, branch, new_base, source_repo_url=None):
+        """Merge `new_base` into `branch` and return the new HEAD commit id.
```
Maybe `source` and `target` instead of `branch` and `new_base`. Or `source_branch` and `target_branch`.

mdevlamynck commented:
I fixed the two wording points you mentioned. I also updated `_fuse_branch()` to remove the use of the word `base` since that function is supposed to be generic over merge and rebase.
